packages/appeals-service-api/src/routes/v2/appeal-cases/repo.js

```diff
@@ -39,17 +82,142 @@ class AppealCaseRepository {
 	 * Get an appeal by case reference (aka appeal number)
 	 *
 	 * @param {string} caseReference
-	 * @param {AppealCaseCreateInput} data
+	 * @param {AppealHASCase} data
 	 * @returns {Promise<AppealCase>}
 	 */
-	putByCaseReference(caseReference, data) {
-		return this.dbClient.appealCase.upsert({
-			create: { ...data, Appeal: { create: {} } }, // create with parent Appeal
-			update: data,
+	async putHASByCaseReference(caseReference, data) {
+		const mappedData = mapHASDataModelToAppealCase(data);
+
+		// todo: move this to mapping
+		delete mappedData.caseType;
+		delete mappedData.linkedCaseStatus;
+		delete mappedData.leadCaseReference;
+		delete mappedData.notificationMethod;
+		delete mappedData.nearbyCaseReferences;
+		delete mappedData.neighbouringSiteAddresses;
+		delete mappedData.affectedListedBuildingNumbers;
```

Comment:
These are handled in separately as they are many-many/lookups 
Initially had these as ignored params in the mapping but linter complained, cleaner suggestions welcome

Reply:
I've gone with a strategy of delete all + add all rather than checking if each exists

Reply:
yep should have remembered that